src/server/storage/cron/CleanupAbandonedSignups.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import logger from '../../../imports/logger'
+import Config from '../../server.config'
+
+class CleanupAbandonedSignups {
+  schedule = null
+  model = null
+  logger = null
+
+  get name() {
+    return 'storage/cleanup_abandoned_signups'
+  }
+
+  constructor(Config, storage, logger) {
+    const { storageCleanupCron } = Config
+
+    this.logger = logger
+    this.schedule = storageCleanupCron
+    this.model = storage.model
+  }
+
+  async execute() {
+    const { model, logger } = this
+
+    const removeCondition = ['torusProvider', 'regMethod', 'loginToken', 'w3Token', 'createdDate'].map(field => ({
```

Comment:
pls look at @sirpy 's comments

i think we need to change condition for 'isCompleted' not exists **or** 'createdDate' not exists **or** 'lastLogin' <= 'createdDate'